site/src/components/NavigationBar/NavigationBar.tsx

```diff
@@ -0,0 +1,147 @@
+ */
+class NavigationBar extends React.Component<Props, unknown> {
+  static displayName = "NavigationBar"
+  static Link = Link
+  static Menu = Menu
+  static defaultProps: Props = {
+    environment: "production",
+    loading: false,
+    colorScheme: "cultureamp",
+    badgeHref: "/",
+  }
+
+  render() {
+    const { children, colorScheme = "cultureamp" } = this.props
+
+    return (
+      <Media query={MOBILE_QUERY}>
+        {(matches: boolean) =>
+          matches ? (
+            <ControlledOffCanvas
+              headerComponent={this.renderBadge()}
+              footerComponent={this.props.footerComponent}
+              links={children}
+              heading="Menu"
+              menuId="menu"
+            />
+          ) : (
+            <header
+              className={classNames(styles.navigationBar, styles[colorScheme])}
+            >
+              <span className={styles.caLogoContainer}>
+                <Brand reversed variant="logo-horizontal" alt="Culture Amp" />
+              </span>
+              {this.renderNav(children)}
+            </header>
+          )
+        }
+      </Media>
+    )
+  }
+
+  renderNav(children?: Navigation) {
+    if (!children) return null
+
+    return (
+      <nav className={styles.links}>
+        {(Object.keys(children) as Array<keyof typeof children>).map(key =>
+          this.renderNavSection(key, children[key] || [])
+        )}
+      </nav>
+    )
+  }
+
+  renderNavSection(section: string, items: NavigationItem[]) {
+    return (
+      <ul
+        key={`${section}-${uuidv4()}`}
+        className={classNames({
+          [styles.primary]: section === "primary",
+          [styles.secondary]: section === "secondary",
+          [styles.final]: section === "final",
+        })}
+      >
+        {items.map(item => this.renderNavItem(item, section))}
+      </ul>
+    )
+  }
+
+  renderNavItem(link: NavigationItem, section: string) {
+    const linkWithSection = { ...link, props: { ...link.props, section } }
+
+    const key =
+      "href" in linkWithSection.props
+        ? linkWithSection.props.href
+        : linkWithSection.props.heading
+
+    return (
+      <li
+        key={`${key}-${uuidv4()}`}
+        className={classNames(styles.child, {
+          [styles.active]: linkWithSection.props.active,
+        })}
+      >
+        {linkWithSection}
+      </li>
+    )
+  }
+
+  renderBadge() {
+    const {
+      environment = "test",
+      loading = false,
+      badgeHref = "/",
+      colorScheme = "kaizen",
+    } = this.props
+
+    const badges = {
+      production: ProductionBadge,
+      staging: StagingBadge,
+      test: TestBadge,
+      local: LocalBadge,
+    }
+
+    const Badge = badges[environment] || namedBadge(environment)
+
+    return (
+      <Badge loading={loading} href={badgeHref} colorScheme={colorScheme} />
+    )
+  }
+}
+
+export default withDeprecatedComponent(NavigationBar, {
```

Comment:
I'd get rid of this `withDeprecatedComponent` HOC wrapper now, it's not relevant here and will just pollute the logs.

Reply:
I will try to remove any application logic that is not related to the Kaizen site directly, as we know that the Culture amp apps has their own composition, where Kaizen will not provide the navigationBar.